routing/router.go

```diff
@@ -1822,19 +1822,32 @@ func (r *ChannelRouter) sendPayment(
 func (r *ChannelRouter) processSendError(paySession PaymentSession,
 	rt *route.Route, fErr *htlcswitch.ForwardingError) bool {
 
-	errSource := fErr.ErrorSource
-	errVertex := route.NewVertex(errSource)
+	failureSourceIdx := fErr.FailureSourceIdx
+	log.Tracef("node=%x reported failure when sending htlc", failureSourceIdx)
 
-	log.Tracef("node=%x reported failure when sending htlc", errVertex)
+	var (
+		failureVertex route.Vertex
+		failureSource *btcec.PublicKey
+	)
+	if failureSourceIdx > 0 {
+		var err error
+		failureVertex = rt.Hops[failureSourceIdx-1].PubKeyBytes
```

Comment:
The failure index passed up from the onion layer isn't zero-indexed? If not IMO it should be.

Reply:
I propose to define index zero as the self node in the route. That indeed means that any failure received over the network will have at least index 1. 

There currently are locally generated failures. They occur even before we send out the network packet and get index zero. How would we deal with those if the onion layer would be zero-indexed?

Using two different definitions of failure index may be confusing.